javalin/src/main/java/io/javalin/http/ErrorMapper.kt

```diff
@@ -15,10 +15,15 @@ class ErrorMapper {
     fun addHandler(statusCode: Int, contentType: String, handler: Handler) =
         errorHandlers.add(MapperEntry(statusCode, contentType, handler))
```

Comment:
Can this be removed?

Reply:
In theory user can define custom status code from valid range.

Reply:
I'd have to refactor the whole ErrorMapper class to use enums instead of ints, i'll push a commit.

Reply:
And alternative would be to make it private/internal?

Reply:
I'm not sure if it's valid call, are you sure you want to remove support for custom status codes? @tipsy

Reply:
@dzikoysk we need to cover that somehow too 🤔

Reply:
I believe the question is whether to allow unknown http codes to be mapped to errors, If they're not I'll just refactor the class to be consistent.

Reply:
I think it's fair to allow users to do that. We can have a marker interface that the enum implements, then people can create their own enum?

Reply:
I'd keep both variants not only because it's convenient, but also simplifies handling if you want to integrate other libraries that more likely return int/custom enum with int codes

Reply:
I really dislike offering two ways of doing the same thing 🤔

Reply:
An interface might be feasble, using `HttpCode#getStatus` and `HttpCode#getMessage`.  But I agree with @dzikoysk , keeping both might be better.

Reply:
I don't want to enable people to write inconsistent apps. Apps with more than 1 person is almost certainly going to end up with a mix of `int` and `HttpCode`, so I only want to support one.

Reply:
Should I then refactor `ErrorMapper` to use only enums?

Reply:
I'm working on a commit using the interface approach, however, internally we should still use Ints for error mapping. With status codes stored in the `res: HttpServletResponse` val there's no way to reconstruct the custom code enum, unless the context stores it too.

Reply:
> internally we should still use Ints for error mapping.

Agreed !

Reply:
Done! Thoughts? The only test failing atm is this one https://github.com/javalin/javalin/pull/1619#issuecomment-1189010679